### ISSUE TITLE
Separate the option to use built-in icons

### ISSUE THREAD

**Issue by bookletchoir** [CDrummond/cantata#792]
I'm using a random stand-alone window manager without any desktop environment, which have heavy usage of gtk+ themes/icons widely.

Before version 2.0, the application will use qt default icons, but I can use this environment variable to make it use gtk+ icons: `DESKTOP_SESSION=gnome`

Since version 2.0, the application started to use built-in icons, makes the icons in tray and the application itself look unified with other applications.

Can you make a separate option whether to use built-in icons or not?

**Comment by CDrummond** [comment]
Try setting Qt/KDE apps to use gnome icon theme, then set DESKTOP_SESSION=kde before starting cantata. Cantata should then use the KDE icon theme.

**Comment by bookletchoir** [comment]
One way to use gtk icons in KDE apps for non-DE is to [set `DESKTOP_SESSION=gnome`](https://wiki.archlinux.org/index.php/Uniform_look_for_Qt_and_GTK_applications#Using_a_GTK.2B_icon_theme_in_Qt_apps), and it worked for cantata v1.5.x and other KDE apps (except some hard-coded ones).

As I'm not using KDE or Gnome, using the native KDE systemsettings won't apply the settings when the application is running outside KDE.

**Comment by CDrummond** [comment]
And? You are not using gnome either. Did you even try the suggestion?

**Comment by CDrummond** [comment]
Anyway, for 2.1, I've added a new config item to set Cantata to use the configured icon theme. This needs to be set by editing the config file - it's not something I think warrants being user visible.

**Comment by bookletchoir** [comment]
To use gtk icons for KDE apps _for non-DE_, it's needed to set `DESKTOP_SESSION` to `gnome` and you said set it back to `kde`, which doesn't make any sense at all.

I _do_ have a proper environment for KDE, I even installed the whole plasma DE to config the settings there, for additional, I had a look at `qt5ct` and `qtconfig-qt4` and set up configs and themes there.

But even when running in plasma DE, its icons is still not as expected, [the play/stop/next/previous icon and folder icon is the same](http://i.imgur.com/MPk21Kc.png), while the icons in app setting seems to applied correctly.

And yes, as soon as I leave the plasma DE and use any wm, using `DESKTOP_SESSION=kde` won't apply gtk icons for cantata, only the theme.

Edit: I'll give a try on new ver if it's solved.

**Comment by CDrummond** [comment]
With Cantata 2.x, setting XDG_CURRENT_DESKTOP=KDE and **not** using breeze will cause it to use the KDE configured icon theme. In 2.0.0 that is the only way to not use Cantata's own icons.

Hence, if Qt was configured to use gnome icons (via qtconfig?) then if Cantata thought it was running under KDE, it'd use the configured (gnome) icons.

I've updated the code, so that the toolbar icons will now also use the standard icons - if the config item is set. However, the statusbar icons, and the 'action' icons in the views are still themed via font-awesome.

Sorry, if you don't like this - but that's the way it is. Trying to get icons correct in every DE is a nightmare. So, I took the stance of just packaging some modern looking icons - and using mono icons elsewhere. Personally, this works for me - so I have not intention of chaging.

**Comment by bookletchoir** [comment]
*(Previous argument, fixed (partly) for now):* I don't know where I got it wrong, but I _did_ install KDE5 (plasma or whatever it is called), set all theme/icon configs I saw to gtk so it will (perhaps) create a proper config files, and it _only works in KDE5_. Despite whatever the env i set in the wm, it won't apply the config to cantata, or because I'm using Qt5 build? I gave another look in qtconfig of both qt4 and qt5, and there's an option for icons for qt5. All set, but it's still not working as expected, what else can I do?

I've been using cantata since version 0.8, not a new user. What I'm complaining is what have been working fine before and used to be, but was missing in newer version. And, it was never a good ideal to make the user create a custom bash file or edit desktop file to make the application runs correctly.

For the new patch, you did fine, but you forgot something, It was not there [when I'm using KDE5](http://i.imgur.com/MPk21Kc.png) either:
![](http://i.imgur.com/JwZyGWQ.png)
Other things work fine.
